File: PhysioHealthCare.Domain/Entities/Appointment.cs
namespace PhysioHealthCare.Domain.Entities
{
    using PhysioHealthCare.Domain.Common;
    using PhysioHealthCare.Domain.Enums;
    using System.Net.NetworkInformation;

    public class Appointment : BaseEntity
    {
        public Guid PatientId { get; set; }

        public DateTime AppointmentDate { get; set; }

        public string Reason { get; set; } = string.Empty;

        public string Notes { get; set; } = string.Empty;

        public AppointmentStatus Status { get; set; }
            = AppointmentStatus.Scheduled;

        public Patient Patient { get; set; } = null!;
    }
}

File: PhysioHealthCare.Web/src/app/shared/models/appointment.ts
export type AppointmentStatus =
  | 'Scheduled'
  | 'InProgress'
  | 'Completed'
  | 'Cancelled';

export type AppointmentStatusValue =
  | 1
  | 2
  | 3
  | 4;

export type AppointmentSortField =
  | 'appointmentDate'
  | 'patientName'
  | 'status'
  | 'reason';

export interface Appointment {
  id: string;
  patientId: string;
  patientName: string;
  appointmentDate: string;
  reason: string;
  notes: string | null;
  status: AppointmentStatus;
}

export interface CreateAppointment {
  patientId: string;
  appointmentDate: string;
  reason: string;
  notes?: string;
}

export interface UpdateAppointment {
  appointmentDate: string;
  reason: string;
  notes?: string;
}

export interface UpdateAppointmentStatus {
  status: AppointmentStatusValue;
}

export interface AppointmentFilters {
  patientId?: string;
  status?: AppointmentStatusValue;
  dateFrom?: string;
  dateTo?: string;
  search?: string;
  sortBy?: AppointmentSortField;
  sortDirection?: 'asc' | 'desc';
}